File: app/src/test/java/com/zl/weilu/androidut/junit/MyRule.java
package com.zl.weilu.androidut.junit;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;

/**
 * 作者：weilu on 2017/10/16 13:23
 */

public class MyRule implements TestRule {

    @Override
    public Statement apply(final Statement base, final Description description) {

        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                // evaluate前执行方法相当于@Before
                String methodName = description.getMethodName(); // 获取测试方法的名字
                System.out.println(methodName + "测试开始！");

                base.evaluate();  // 运行的测试方法

                // evaluate后执行方法相当于@After
                System.out.println(methodName + "测试结束！");
            }
        };
    }

}

File: app/src/test/java/com/zl/weilu/androidut/junit/MyRule.kt
package com.zl.weilu.androidut.junit

import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement

/**
 * 作者：weilu on 2017/10/16 13:23
 */

class MyRule : TestRule {

    override fun apply(base: Statement, description: Description): Statement {

        return object : Statement() {
            @Throws(Throwable::class)
            override fun evaluate() {
                // evaluate前执行方法相当于@Before
                val methodName = description.methodName // 获取测试方法的名字
                println(methodName + "测试开始！")

                base.evaluate()  // 运行的测试方法

                // evaluate后执行方法相当于@After
                println(methodName + "测试结束！")
            }
        }
    }

}
